src/Catalog/Monitoring/Validation/Test/ValidationTestRunner.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Protocol.Core.Types;
+using System.Threading.Tasks;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Result;
+using NuGet.Services.Metadata.Catalog.Monitoring.Providers;
+using static NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test.ValidationTestUtility;
+using NuGet.Protocol;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    /// <summary>
+    /// Performs a validation test on a <see cref="ValidationTestRunData"/>.
+    /// </summary>
+    public interface IValidationTestRunner
+    {
+        /// <summary>
+        /// Returns whether or not the validation test should run on a <see cref="ValidationTestRunData"/>.
+        /// </summary>
+        Task<bool> ShouldRun(ValidationTestRunData data);
```

Comment:
ShouldRun is never used outside ValidationTestRunner , why should it be part of the interface?